Snowflake.Data.Tests/IntegrationTests/SFPutGetTest.cs

```diff
@@ -496,23 +521,51 @@ private void GetFile(DbConnection conn)
                     reader.GetString((int)SFResultSet.PutGetResponseRowTypeInfo.ResultStatus));
 
                 // Check file contents
-                using (var streamReader = new StreamReader(t_outputFilePath))
+                foreach (var line in ReadOutputFileLinesUnzippingIfCompressedBecauseOfAutoCompression())
                 {
-                    while (!streamReader.EndOfStream)
+                    if (string.IsNullOrEmpty(line)) continue;
+                    var values = line.Split(',');
+
+                    for (var i = 0; i < s_colData.Length; i++)
                     {
-                        var line = streamReader.ReadLine();
-                        if (line == null) continue;
-                        var values = line.Split(',');
-
-                        for (var i = 0; i < s_colData.Length; i++)
-                        {
-                            Assert.AreEqual(s_colData[i], values[i]);
-                        }
+                        Assert.AreEqual(s_colData[i], values[i]);
                     }
                 }
             }
         }
 
+        private static string[] ReadOutputFileLinesUnzippingIfCompressedBecauseOfAutoCompression()
+        {
+            using (var outputStream = File.OpenRead(t_outputFilePath))
+            {
+                if (!IsCompressedBecauseOfAutoCompression())
+                {
+                    return splitLines(outputStream);
+                }
+                using (var decompressingStream = new GZipStream(outputStream, CompressionMode.Decompress))
+                {
+                    using (var decompressedStream = new MemoryStream())
+                    {
+                        decompressingStream.CopyTo(decompressedStream);
+                        return splitLines(decompressedStream);
+                    }
+                }
+            }
+        }
+
+        private static string[] splitLines(Stream stream)
+        {
+            var bytes = new byte[stream.Length];
+            stream.Position = 0;
+            stream.Read(bytes, 0, (int) stream.Length);
+            return Encoding.UTF8.GetString(bytes).Split('\n');
+        }
+
+        private static bool IsCompressedBecauseOfAutoCompression()
+        {
+            return t_compressionType == "none" && t_destCompressionType == "gzip" && t_autoCompress;
```

Comment:
it's not clear how "AutoCompression" name relates to both names - compressionType and destCompressionType. None of them is having a notion of auto compression. Can we make it more self explanatory, either variables and in the end have just: `IsCompressed` method?

Reply:
replaced by IsCompressedByTheDriver()